plugins/woocommerce/tests/api-core-tests/global-setup.js

```diff
@@ -1,8 +1,200 @@
+			'Plugins'
+		);
+
+		console.log( 'Deactivating WooCommerce Plugin...' );
+		await setupPage.locator( '#deactivate-woocommerce' ).click();
+		await expect( setupPage.locator( 'div#message' ) ).toHaveText(
+			'Plugin deactivated.Dismiss this notice.'
+		);
+
+		console.log( 'Deleting WooCommerce Plugin...' );
+		setupPage.on( 'dialog', ( dialog ) => dialog.accept() );
+		await setupPage.locator( '#delete-woocommerce' ).click();
+		await expect( setupPage.locator( '#woocommerce-deleted' ) ).toHaveText(
+			'WooCommerce was successfully deleted.'
+		);
+
+		for ( let i = 0; i < adminRetries; i++ ) {
+			try {
+				console.log( 'Reinstalling WooCommerce Plugin...' );
+				await setupPage.goto( 'wp-admin/plugin-install.php' );
+				await setupPage.locator( 'a.upload-view-toggle' ).click();
+				await expect(
+					setupPage.locator( 'p.install-help' )
+				).toBeVisible();
+				await expect(
+					setupPage.locator( 'p.install-help' )
+				).toContainText(
+					'If you have a plugin in a .zip format, you may install or update it by uploading it here.'
+				);
+				const [ fileChooser ] = await Promise.all( [
+					setupPage.waitForEvent( 'filechooser' ),
+					setupPage.locator( '#pluginzip' ).click(),
+				] );
+				await fileChooser.setFiles( woocommerceZipPath );
+				console.log( 'Uploading nightly build...' );
+				await setupPage
+					.locator( '#install-plugin-submit' )
+					.click( { timeout: 60000 } );
+				await setupPage.waitForLoadState( 'networkidle' );
+				await expect(
+					setupPage.getByRole( 'link', { name: 'Activate Plugin' } )
+				).toBeVisible();
+				console.log( 'Activating Plugin...' );
+				await setupPage
+					.getByRole( 'link', { name: 'Activate Plugin' } )
+					.click( { timeout: 60000 } );
+				pluginActive = true;
+				break;
+			} catch ( e ) {
+				console.log(
+					`Installing and activating plugin failed, Retrying... ${ i }/${ adminRetries }`
+				);
+				console.log( e );
+			}
+		}
+		if ( ! pluginActive ) {
+			console.error(
+				'Cannot proceed api test, as installing WC failed. Please check if the test site has been setup correctly.'
+			);
+			process.exit( 1 );
+		}
+
+		console.log( 'WooCommerce Re-installed.' );
+		await expect(
+			setupPage.getByRole( 'heading', { name: 'Welcome to Woo!' } )
+		).toBeVisible();
+
+		await deleteZip( woocommerceZipPath );
+
+		// Might need to update the database
+		await setupPage.goto( 'wp-admin/plugins.php' );
+		const updateButton = setupPage.locator(
+			'text=Update WooCommerce Database'
+		);
+		const updateCompleteMessage = setupPage.locator(
+			'text=WooCommerce database update complete.'
+		);
+		await expect( setupPage.locator( 'div.wrap > h1' ) ).toHaveText(
+			'Plugins'
+		);
+		if ( await updateButton.isVisible() ) {
+			console.log( 'Database update button present. Click it.' );
+			await updateButton.click( { timeout: 60000 } );
+			await expect( updateCompleteMessage ).toBeVisible();
+		} else {
```

Comment:
Once the update is done, we might need to run a DB update.  So check for and do that if required.